Merged PR, removed SCSS warnings, added rows, columns, :user-valid, labels to groups and more

Yohn opened the pull request:
I wasnt sure if this was still maintained and added some helpful resourced I created, and merged a bunch of open pull requests that corrected some issues. 

[Heres my Fork](https://github.com/Yohn/PicoCSS) and [Heres an updated Demo](https://yohn.github.io/PicoCSS/)

y2kbugger committed:
Add date to list of excluded input types.

Zurga committed:
fix: Card in modal does not use modal styling

lukas-kasticky committed:
Allow use of card component with `role="article"`

feored committed:
fix: don't count hidden rows in table stripes

panda-madness committed:
Update _tooltip.scss

nstringham committed:
refactor: move nested declarations after non-nested declarations

see https://sass-lang.com/documentation/breaking-changes/mixed-decls/

fix #577

nstringham committed:
chore: build

preist committed:
Fix several typos throughout the project

mwargan committed:
Fix details inside drop-down being rendered as dropdown

Given a dropdown that contains an accordion within it, the accordion renders as a dropdown because the summary selector also selects the children of a dropdown element. This fixes that so it renders as expected.

```
<details class="dropdown">
   <summary> Select a phase of matter... </summary>
   <ul>
      <li>
         <details>
            <summary> This is a dropdown option </summary>
            <ul>
               <li><label><input type="radio" name="phase" value="gas"> Gas </label></li>
               <li><label><input type="radio" name="phase" value="plasma"> Plasma </label></li>
            </ul>
         </details>
      </li>
   </ul>
</details>
```

coezbek committed:
Update _loading.scss not to affect form elements itself

Yohn committed:
Merge pull request #5 from nstringham/mixed-sass-declarations

Mixed sass declarations

Yohn committed:
Merge pull request #10 from Zurga/fix/modal_with_card

fix: Card in modal does not use modal styling

Yohn committed:
Merge pull request #9 from y2kbugger/main

Add date to list of excluded input types.

Yohn committed:
Merge pull request #7 from feored/fix-striped-table

fix: don't count hidden rows in table stripes

Yohn committed:
Merge pull request #3 from coezbek/main

Update _loading.scss not to affect form elements itself

Yohn committed:
Merge pull request #2 from preist/fix-typos

Fix several typos throughout the project

Yohn committed:
Merge pull request #4 from mwargan/patch-1

Fix details inside drop-down being rendered as dropdown

Yohn committed:
Merge pull request #8 from panda-madness/patch-1

Update _tooltip.scss

Yohn committed:
Merge branch 'main' into pr/Yohn/6

Yohn committed:
closes Yohn/PicoCSS/pull/1 and Yohn/PicoCSS/pull/11

Yohn committed:
surpressimg sass warnings

Yohn committed:
Merge pull request #13 from Yohn/Surpressing-sass-warnings

Surpressing sass warnings and closing 1 & 11

Yohn committed:
Added Notifications 12

Yohn committed:
Finally stopped all the sass compile warnings, changed the tab-size to 2, and added responsive .row, .col-*, and .offset-* classes, using display:gid, the same way bootsrap-5 is set up for the .row, .com-* and .offset* classes.

Yohn committed:
Added <label> compatibility to groups

Yohn committed:
Created build-dev script to only build 1 css file for quicker testing enviroment. Added accordion javascript for slide effect. modified the notification scss, and added more demos.

Yohn committed:
Added a Timeline feature, removed the margin-bottom from `article > header > h[1-6]` And added the border-color for dark theme on the article > header

Yohn committed:
Added form validation

Yohn committed:
Got the file validation and previewer lines up nicer

Yohn committed:
V. 2.1.0 - Yohns Fork

Yohn committed:
Update css urls

Yohn committed:
Enabled multiline tooltips, and disabled dotted lines for tooltips when child elements contain links of form elements.

rcoup committed:
Support web components/shadow DOM.

Add :host to all uses of :root

Yohn committed:
Merge pull request #15 from lukas-kasticky:dev

Allow use of card component with `role="article"`

Yohn committed:
Merge pull request #16 from rcoup/web-components-host

Support web components/shadow DOM.

Yohn committed:
Updated Docs

Yohn committed:
Remove pointer cursor #14

Yohn committed:
rebuilt css files, For some reason was having issues with the line endings, hopefully the editorconfig will fix it.

Yohn committed:
Changed spacing on form elements to be a tad smaller. It was just too big for my liking, and I feel this way is a lot easier.
Added cursor back on button - it was just an arrow.
Added parent-selector to the notifications, and corrected readbility of notifications.
updated the version within the css files
made rows only show up on css files where classes are enabled
made the tooltips have a dark background - even for dark themes.

Yohn committed:
Update README.md

Yohn committed:
Update README.md

Yohn committed:
- Updated form validation to not validate when form[novalidate]
- Updated tooltips to go multiline, max width of 250px.
-Updated card footer, to remove the margin from the last element if they are buttons or groups when in the footer
- Made container, grid, and row classes to have a parent class, so they will not effect rest of page. This will only work if you use that style css.

- Added tabs
- Added floating labels
- Added responsive nav hamburger menu

Let me know if you find any bugs or have ideas for improvements!

Yohn committed:
Update index.html

Make hamburger nav show under xxl viewport
replaced spaces with tabs.

remohammadi committed:
Support RTL for group component border radius styles

remohammadi committed:
Add RTL support for Accordion Marker

Yohn committed:
Merge pull request #18 from remohammadi/main

Yohn committed:
Merged RTL pull request into css files, and updated docs for hamburger menu.
Will create new versioning soon, I have something else I'd like to include.

Yohn committed:
Improved visibility of loading icon in light mode on contrast backgrounds.

mperham commented:
Really nice work. @lucaslarroche would you consider adding additional maintainers like @Yohn to PicoCSS? I’d like to use Pico, I really like the ethos and design but the lack of any maintenance makes it a non-starter.

mperham commented:
I grab the raw CSS from `css/` and adjust a few custom properties. I prefer no build in my project and don’t use npm at all.

My main worry right now is RTL fixes as I support multiple RTL languages like Arabic and Hebrew. Did you pull in those PRs?

Yohn commented:
@mperham yes I did pull in the rtl fixes here, and had another contributor submit another rtl recently.

And the css files in the css directory in my fork have all been updated as well

Yohn committed:
RTL Dropdown Nav corrected thanks  to @64bitint
Added popovers to possibly replace the notifications.
Started adding popover toasts

Yohn committed:
tweaks to safari bug with nav dropdowns

Yohn committed:
sass is removing the "\200B" value and leaving it blank? I put a space there for now..

Yohn committed:
sass compiler was removing the value: "\200B" for the :where(navs li).
Updated, sass to include the 0 width space for VoiceOver, and removed the float for safari  trimming dropdodwn list bugs.

Yohn committed:
Final iOS 18.2 bug fix for [dir=rtl] dropdowns
Added Azure to theme build list

Yohn committed:
corrected dropdown arrow colors for secondary, contrast, and primary. Also enabled the turning of the arrow when the dropdowns are open or closed

Yohn committed:
Corrected Dropdown > Submenu formatting and alignment

Yohn committed:
Cards keep style with form tag wrapped around header and footer in article

Yohn committed:
Version 2.2.1

Yohn committed:
Update CDN CSS url to use 2.2.1

Yohn commented:
# Version 2.2.1

## What's Changed
 - Improve RTL support by @remohammadi in Yohn/PicoCSS#18 
 - Added Popovers - removed Notifications
 - Added a new Tab component Closes Yohn/PicoCSS#20 - removed the old tab component (Old tab component did not work in Google Chrome)
    - Valid HTML, 100% accessible, with left and right arrows to navigate the tabs.
    - uses `[role=tablist]` `[role=tab]` and `[role=tabpanel]` attributes.
 - includes pico.fluid.classless.conditional.min.css #645 file missing
 - Improved Dropdowns and Submenus #116 #543 component
 - Corrected dropdown and accordion arrows to function as expected
 - Corrected button colors for light and dark mode
 - Corrected Loading icon in contrast buttons #643 for dark and light mode
 - Safari RTL dropdown #634 text cut off bug corrected
 - Modal: card > form #616 corrected cards to have a form tag before the header and footer and not lose the formatting.
 - Removed margin from last child #557 in forms for better spacing.
 - increased font size for floating labels 
 - other bug fixes


**Full Changelog**: https://github.com/Yohn/PicoCSS/compare/2.2.0...2.2.1
### New Contributors
* @remohammadi made their first contribution in https://github.com/Yohn/PicoCSS/pull/18

Yohn committed:
Removed unused docs and docs/js files

Yohn committed:
Create node.js.yml

Yohn committed:
Update and rename .github/workflows/node.js.yml to .github/auto-build-maybe.yml.txt

Yohn committed:
Create GitHub Action to BuildCSS.yml

Github action automatically builds the CSS files in the `css/` directory from the `scss/` directory in merge and push requests using npm

github-actions[bot] committed:
Build CSS

Yohn committed:
Create FUNDING.yml

github-actions[bot] committed:
Build CSS

Yohn committed:
Update BuildCSS.yml

Run build script only when change is made within the `scss` directory

Yohn committed:
Closes Legend element not styled in `role="group"` Yohn/PicoCSS#23
Added compatability with `.dropdown` to the `role=group`
Put a dropdown submenu in the group examples
Improved borders on `[role=group] > label, [role=group] > legend` to ensure the same size as other group elements

github-actions[bot] committed:
Build CSS

remohammadi committed:
Fix RTL margin handling in group component

I had made a mistake previously by putting the RTL margin styles one level higher than where it belonged.

github-actions[bot] committed:
Build CSS

Yohn committed:
Merge pull request #25 from remohammadi/main

Fix RTL margin handling in group component

github-actions[bot] committed:
Build CSS

Yohn committed:
Reverted back to before prettier
Better floating label handling
organized validation for better scss
Updated copyright for 2025

github-actions[bot] committed:
Build CSS

Yohn committed:
Updated tabs to work with multiple tabs on same page and nested tabs
Updated modal to be able to close other modals when opening a second modal
updated group in docs

Yohn committed:
Merge branch 'main' of github.com:Yohn/PicoCSS

Yohn committed:
New switch light / dark mode toggle

github-actions[bot] committed:
Build CSS

Yohn committed:
Closes #32 and brings the arrow back to the tooltips

github-actions[bot] committed:
Build CSS

Yohn committed:
Closes #30 dicussion as modal sizes have now been added with this commit.

Cleaned up the submenu within the dropdown some more.

github-actions[bot] committed:
Build CSS

Yohn committed:
2.2.3

github-actions[bot] committed:
Build CSS

Yohn committed:
enhanced the tooltip to have a width of `max-content` and a max width of 250px. This will allow for the tooltips to not always be 250px wide even when the tooltip is short.

Added the ghost buttons from discussion #29

github-actions[bot] committed:
Build CSS

irfanfadilah committed:
Improve Inline Code Spacing

irfanfadilah committed:
Fix Pre Code Overflow

Yohn committed:
Update CONTRIBUTING.md

Yohn committed:
Merge pull request #36 from irfanfadilah/yohn/improve-code-rendering

Improve Code Rendering

github-actions[bot] committed:
Build CSS

Yohn committed:
Only make first level children of tablist button[role="tab"]'s styles for tabs and not have it apply to other buttons within it.

github-actions[bot] committed:
Build CSS

Yohn committed:
Striped table rows updated

github-actions[bot] committed:
Build CSS

Yohn committed:
Added submenus to work with hamburger menu

github-actions[bot] committed:
Build CSS

Yohn committed:
Closes [BUG] Striped table does not work on minified pico css version Yohn/PicoCSS#42
Closes [BUG] Broken burger navbar with dropdown subnav Yohn/PicoCSS#37

2.2.5

github-actions[bot] committed:
Build CSS

Yohn committed:
2.2.6 - cleancss fix for minifying error on on the striped tables #42

github-actions[bot] committed:
Build CSS

Yohn committed:
Update form ID for validation and add new entry to .gitignore

Yohn committed:
Removed the pull request workflow

Pull requests can push to main, so do not have the pull request workflow in here.

Yohn committed:
Updated the clean-css filter to ignore the `odd of` lines so it doesnt create an empty selector

github-actions[bot] committed:
Build CSS

Yohn committed:
Fixes header and footer in modal when form is direct child of the article, and if the form would be between the header and footer tags.
Modals with the size classes on the dialog were not sizing correctly when form was a child of the modal, and this has been corrected.

Yohn committed:
Merge pull request #45 from Yohn/Modal-forms

Modal sizes and styling with form child

github-actions[bot] committed:
Build CSS

Yohn committed:
Prevent the FileValidator's remove button from bubbling up and closing modals when its within a modal.

Yohn committed:
Merge pull request #46 from Yohn/FileValidator-remove-file-bubbling-up

Prevent the FileValidator's remove button from bubbling up and closing modals when its within a modal.

Yohn committed:
Floating label was moving on the textareas when the textareas grow

Yohn committed:
Merge pull request #48 from Yohn/textarea-floating-labels

Textarea floating labels

github-actions[bot] committed:
Build CSS

Yohn committed:
had to update the striped table rows again cause they were failing, and update the floating labels

github-actions[bot] committed:
Build CSS

Yohn committed:
Update README.md

Added other components I've created to the Readme

Yohn committed:
Update README.md

Yohn committed:
Closes Hamburger Menu show part of menu below Yohn/PicoCSS#49
Fixes Accordion chevron padding picocss/pico#653

Demo has been updated with new hamburger nav  example.

github-actions[bot] committed:
Build CSS

Leftium committed:
Fix missing padding on .container.

Fixes https://github.com/picocss/pico/issues/654.

Yohn committed:
Merge pull request #55 from Leftium/container-padding

Fix missing padding on .container.

github-actions[bot] committed:
Build CSS

Yohn committed:
Add padding to container for all viewports

github-actions[bot] committed:
Build CSS

Yohn committed:
updated accordian and dropdown to not overlap the arrow, provide enough space for the arrow, and keep the arrow at the top when theres multi-lines for the sumary text.
Reverted back the container padding.

github-actions[bot] committed:
Build CSS

Yohn committed:
Typo fixed in docs for the table
Added basic padding and margin classes: 0 for none, 1 for half the spacing size, 2 for normal spacing, and 3 for 1.5 spacing
Minor clean up on the accordion.scss

github-actions[bot] committed:
Build CSS

Yohn committed:
Closes Data-theme global setting not work Yohn/PicoCSS#52
Closes Dropdown arrow position issue - v2.2.9 Yohn/PicoCSS#53

Releases for 2.2.10

github-actions[bot] committed:
Build CSS

randolf committed:
Update README.md

Minor grammatical improvements.

Yohn committed:
Update README.md

Yohn committed:
Merge pull request #65 from randolf/patch-2

Update README.md
